src/connector/src/sink/deltalake.rs

```diff
@@ -0,0 +1,544 @@
+                .ok_or_else(|| SinkError::DeltaLake("cannot find field type".to_string()))?;
+            if !check_field_type(&field.data_type, deltalake_field_type)? {
+                return Err(SinkError::DeltaLake(format!(
+                    "column {} type is not match, deltalake is {:?}, rw is{:?}",
+                    field.name, deltalake_field_type, field.data_type
+                )));
+            }
+        }
+        Ok(())
+    }
+
+    async fn new_coordinator(&self) -> Result<Self::Coordinator> {
+        Ok(DeltaLakeSinkCommitter {
+            table: self.config.common.create_deltalake_client().await?,
+        })
+    }
+}
+
+impl TryFrom<SinkParam> for DeltaLakeSink {
+    type Error = SinkError;
+
+    fn try_from(param: SinkParam) -> std::result::Result<Self, Self::Error> {
+        let config = DeltaLakeConfig::from_hashmap(param.properties.clone())?;
+        DeltaLakeSink::new(config, param)
+    }
+}
+
+pub struct DeltaLakeSinkWriter {
+    pub config: DeltaLakeConfig,
+    schema: Schema,
+    pk_indices: Vec<usize>,
+    writer: RecordBatchWriter,
+    is_append_only: bool,
+    dl_schema: Arc<deltalake::arrow::datatypes::Schema>,
+    dl_table: DeltaTable,
+}
+
+impl DeltaLakeSinkWriter {
+    pub async fn new(
+        config: DeltaLakeConfig,
+        schema: Schema,
+        pk_indices: Vec<usize>,
+        is_append_only: bool,
+    ) -> Result<Self> {
+        let dl_table = config.common.create_deltalake_client().await?;
+        let writer = RecordBatchWriter::for_table(&dl_table)?;
+        let dl_schema: Arc<deltalake::arrow::datatypes::Schema> =
+            Arc::new(covert_schema(dl_table.get_schema()?)?);
+
+        Ok(Self {
+            config,
+            schema,
+            pk_indices,
+            writer,
+            is_append_only,
+            dl_schema,
+            dl_table,
+        })
+    }
+
+    async fn write(&mut self, chunk: StreamChunk) -> Result<()> {
+        let a = to_record_batch_with_schema(self.dl_schema.clone(), &chunk)
+            .map_err(|err| SinkError::DeltaLake(format!("convert record batch error: {}", err)))?;
+        self.writer.write(a).await?;
+        Ok(())
+    }
+}
+
+fn covert_schema(schema: &StructType) -> Result<deltalake::arrow::datatypes::Schema> {
+    let mut builder = deltalake::arrow::datatypes::SchemaBuilder::new();
+    for field in schema.fields() {
+        let dl_field = deltalake::arrow::datatypes::Field::new(
+            field.name(),
+            deltalake::arrow::datatypes::DataType::try_from(field.data_type())
+                .map_err(|err| SinkError::DeltaLake(format!("convert schema error: {}", err)))?,
+            field.is_nullable(),
+        );
+        builder.push(dl_field);
+    }
+    Ok(builder.finish())
+}
+
+#[async_trait]
+impl SinkWriter for DeltaLakeSinkWriter {
+    type CommitMetadata = Option<SinkMetadata>;
+
+    async fn write_batch(&mut self, chunk: StreamChunk) -> Result<()> {
+        if self.is_append_only {
```

Comment:
We can just simply call `self.write` and only report error when we meet row op of update or delete.